### PR TITLE
morewaita-icon-theme: init at 42.3

### PR DESCRIPTION
**Pull request by pkosel**
###### Description of changes

Add [MoreWaita](https://github.com/somepaulo/MoreWaita), an Adwaita style extra icons theme for Gnome Shell.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/morewaita-icon-theme/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    cp -r . $out/share/icons/MoreWaita
+    gtk-update-icon-cache -f -t $out/share/icons/MoreWaita && xdg-desktop-menu forceupdate
+  '';
+
+  meta = with lib; {
+    description = "An Adwaita style extra icons theme for Gnome Shell.";
+    homepage = "https://github.com/somepaulo/MoreWaita";
+    license = with licenses; [gpl3];
+    platforms = platforms.linux;
+    maintainers = with maintainers; [pkosel];
```
Comment (wegank) [inline pkgs/data/icons/morewaita-icon-theme/default.nix:36]: Please add yourself to the maintainer list in a separate commit before this one.

File: pkgs/data/icons/morewaita-icon-theme/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  installPhase = ''
+    install -d $out/share/icons/MoreWaita
+    rm -rf ./.git
+    cp -r . $out/share/icons/MoreWaita
+    gtk-update-icon-cache -f -t $out/share/icons/MoreWaita && xdg-desktop-menu forceupdate
+  '';
+
+  meta = with lib; {
+    description = "An Adwaita style extra icons theme for Gnome Shell.";
```
Comment (wineee) [inline pkgs/data/icons/morewaita-icon-theme/default.nix:32]:
```suggestion
    description = "An Adwaita style extra icons theme for Gnome Shell";
```

File: pkgs/data/icons/morewaita-icon-theme/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    homepage = "https://github.com/somepaulo/MoreWaita";
+    license = with licenses; [gpl3];
```
Comment (wineee) [inline pkgs/data/icons/morewaita-icon-theme/default.nix:34]:
```suggestion
    license = with licenses; [ gpl3Only ];
```

File: pkgs/data/icons/morewaita-icon-theme/default.nix (hunk 1 above)
Comment (wineee) [inline pkgs/data/icons/morewaita-icon-theme/default.nix:36]:
```suggestion
    maintainers = with maintainers; [ pkosel ];
```

File: pkgs/data/icons/morewaita-icon-theme/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  nativeBuildInputs = [
+    gtk3
+    xdg-utils
+  ];
+
+  installPhase = ''
+    install -d $out/share/icons/MoreWaita
+    rm -rf ./.git
```
Comment (wineee) [inline pkgs/data/icons/morewaita-icon-theme/default.nix:26]:
```suggestion
```
It is not necessary to do this，`.git` will not install to $out

**Comment by AnhQuanTrl** [comment]
Can this PR be merged?

**Comment by wegank** [comment]
This PR has merge conflicts. I'll try to rebase and force push, but pushing a full year's commits is going to take a long time.

**Comment by dch82** [comment]
Thanks guys! Really appreciate this!

**Pull request merged**